pkg/virt-handler/vm.go

```diff
@@ -1380,6 +1404,45 @@ func (d *VirtualMachineController) handleMigrationProxy(vmi *v1.VirtualMachineIn
 	return nil
 }
 
+func (d *VirtualMachineController) migrateNetworkCacheFiles(vmi *v1.VirtualMachineInstance) error {
+	res, err := d.podIsolationDetector.Detect(vmi)
+	if err != nil {
+		return fmt.Errorf("failed to detect isolation for launcher pod: %v", err)
+	}
+
+	// migrate cache network files from old location private to virt-launcher into
+	// shared location managed by virt-handler; we cannot switch namespace
+	// for the handler process because it's multi-threaded, and instead use
+	// /proc/<pid>/root interface to access to the filesystem view of the
+	// launcher process
+	oldroot := fmt.Sprintf("/proc/%d/root/var/run/kubevirt-private", res.Pid())
+	files, err := ioutil.ReadDir(oldroot)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+		return fmt.Errorf("failed to read old network cache path from launcher pod: %v", err)
+	}
+	for _, info := range files {
+		if info.IsDir() {
+			continue
+		}
+		oldpath := path.Join(oldroot, info.Name())
+		if filepath.Ext(oldpath) != ".json" {
```

Comment:
... nor in non-JSON files.

Could you make the check positive?

If IsFile and IsJSON // explanation why we care only about these